### ISSUE TITLE
Why synchronized methods are removed in recent versions

### ISSUE THREAD

**Issue by jiaz** [kesla/node-snappy#49]
Hi. Thanks for the fantastic library.
We adopted the 2.1 version in our production code. Recently I noticed that that in the latest version, the synchronized implementations are removed. Is there any rational for this change?
I noticed that the Async work is scheduled on the main event loop, since node is single thread app, it looks like there is no much benefit making the computational heavy work async.

I wrote a test to benchmark the performance of async method vs sync method. The code is something like below. (I used 2.1 version node-snappy)

```
var snappy = require('snappy');
var Benchmark = require('benchmark');
var fs = require('fs');

var suite = new Benchmark.Suite;

var data = fs.readFileSync('./MOCK_DATA.json', {encoding: 'utf-8'});
var compressed = snappy.compressSync(data);

suite.add('snappy#compress',{
    'defer': true,
    'fn': function comporess(deferred) {
        snappy.compress(data, function(err, d) {
            deferred.resolve();
        });
    }
})
.add('snappy#compressSync', function() {
    snappy.compressSync(data);
})
.add('snappy#uncompress', {
    'defer': true,
    'fn': function uncompress(deferred) {
        snappy.decompress(compressed, null, function(err, result) {
            deferred.resolve();
        });
    }
})
.add('snappy#uncompressSync', function() {
    snappy.uncompressSync(compressed);
})
.on('cycle', function(event) {
    console.log(String(event.target));
})
.on('complete', function() {
    console.log('Fastest is ' + this.filter('fastest').pluck('name'));
})
.run();
```

the result looks like:

snappy#compress x 507 ops/sec ±1.74% (48 runs sampled)
snappy#compressSync x 3,187 ops/sec ±3.74% (90 runs sampled)
snappy#uncompress x 634 ops/sec ±2.11% (40 runs sampled)
snappy#uncompressSync x 12,406 ops/sec ±4.71% (85 runs sampled)

Looks like it's better to just make the function just synchronized. Do you have any reason to support changing to asynchronized? Thanks!

**Comment by kesla** [comment]
Hey,

Yes, running a method asynchronous is a little bit slower - but you're not blocking the event loop when you do the computation, that's the reason the sync methods were removed.

However, since the focus on this module is speed and since the synchronous methods are that much faster it might make sense to reconsider the decision to remove the synchronous methods - I'd gladly accept a PR including them again if someone has bandwidth!

**Comment by kesla** [comment]
I'm working on this. Almost ready

**Comment by jiaz** [comment]
@kesla Thanks for your quick response :)

**Comment by kesla** [comment]
@jiaz Well, I'm happy if anyone's using this :)

**Comment by danielsantiago** [comment]
Is there any update about this? Thanks

**Comment by kesla** [comment]
hey @danielsantiago - I started working on it in #50 but I got some weird failing tests that I couldn't really wrap my head around. Any help would be very appreciated.

**Comment by jiaz** [comment]
@kesla I submitted a PR to fix the sync implementation. please check.

**Comment by kesla** [comment]
sync versions released as part of 3.1.0!
